common/scala/src/main/scala/whisk/http/BasicHttpService.scala

```diff
@@ -86,39 +86,57 @@ trait BasicHttpService extends HttpService with TransactionCounter {
      */
     def loglevelForRoute(route: String): Logging.LogLevel = Logging.InfoLevel
 
+    /** Rejection handler to terminate connection on a bad request. Delegates to Akka handler. */
+    val prioritizeRejections = recoverRejections { rejections =>
+        val priorityRejection = rejections.find {
+            case rejection: UnacceptedResponseContentTypeRejection => true
+            case _ => false
+        }
+
+        priorityRejection.map(rejection => Rejected(Seq(rejection))).getOrElse(Rejected(rejections))
+    }
+
+    val toStrict = mapInnerRoute { innerRoute =>
```

Comment:
Looks like `toStrictEntity` will work. As for as the timeout goes, the directive will fail if the stream doesn't complete in that about alotted time.

Ran into an error when creating an action with a zip file greater than 1 mb. This is the error message `Substream Source cannot be materialized more than once` There seems to be a bug in akka-http causing the problem, but `toStrictEntity` is being used as a work around right now. Found this info about the error: https://github.com/akka/akka-http/issues/73.

https://github.com/akka/akka-http/issues/73